### PR TITLE
Feature: packet documentation

### PR DESCRIPTION
**Pull request by ImNotAVirus**

### CODE REVIEW

File: apps/elven_gard/lib/elven_gard/helpers/packet.ex [hunk 1]
```diff
@@ -4,19 +4,29 @@ defmodule ElvenGard.Helpers.Packet do
   by an `ElvenGard.Helpers.Frontend`
   """
 
+  alias ElvenGard.Structures.{FieldDocumentation, PacketDocumentation}
```
Comment (Shikanime) [inline apps/elven_gard/lib/elven_gard/helpers/packet.ex]: Why are you mixing runtime and compile structures ?

File: apps/elven_gard/lib/elven_gard/helpers/packet.ex [hunk 2]
```diff
@@ -26,23 +36,31 @@ defmodule ElvenGard.Helpers.Packet do
...
   Define a new packet
   """
-  defmacro packet(packet_type, do: exp) do
-    def_packet(__CALLER__.module, packet_type)
+  defmacro packet(packet_name, do: exp) do
+    def_packet(__CALLER__.module, packet_name)
 
     quote do
+      desc = @desc
+      @elven_current_packet PacketDocumentation.new(unquote(packet_name), desc)
```
Comment (Shikanime) [inline apps/elven_gard/lib/elven_gard/helpers/packet.ex]: Remove this variable

File: apps/elven_gard/lib/elven_gard/helpers/packet.ex [hunk 3]
```diff
@@ -52,11 +70,19 @@ defmodule ElvenGard.Helpers.Packet do
 
   ¯\_(ツ)_/¯
   """
-  defmacro useless_packet(packet_type) do
+  defmacro useless_packet(packet_name) do
+    caller = __CALLER__.module
+
     quote do
+      desc = @desc
```
Comment (Shikanime) [inline apps/elven_gard/lib/elven_gard/helpers/packet.ex]: Remove this variable too

File: apps/elven_gard/lib/elven_gard/helpers/packet.ex (hunk 2 above)
Comment (ImNotAVirus) [reply apps/elven_gard/lib/elven_gard/helpers/packet.ex]: Will be done in the next commit

File: apps/elven_gard/lib/elven_gard/helpers/packet.ex (hunk 3 above)
Comment (ImNotAVirus) [reply apps/elven_gard/lib/elven_gard/helpers/packet.ex]: Will be done in the next commit

**Pull request merged**